collector/src/compare.rs

```diff
@@ -7,16 +7,87 @@ use database::{
 };
 use tabled::{Table, Tabled};
 
+static ALL_METRICS: &[Metric] = &[
+    Metric::InstructionsUser,
+    Metric::Cycles,
+    Metric::WallTime,
+    Metric::MaxRSS,
+    Metric::LinkedArtifactSize,
+    Metric::AssemblyFileSize,
+    Metric::BranchMisses,
+    Metric::CacheMisses,
+    Metric::CodegenUnitLlvmIrCount,
+    Metric::CodegenUnitSize,
+    Metric::ContextSwitches,
+    Metric::CpuClock,
+    Metric::CpuClockUser,
+    Metric::CrateMetadataSize,
+    Metric::CyclesUser,
+    Metric::DepGraphSize,
+    Metric::DocByteSize,
+    Metric::DwoFileSize,
+    Metric::Faults,
+    Metric::FaultsUser,
+    Metric::LlvmBitcodeSize,
+    Metric::LlvmIrSize,
+    Metric::ObjectFileSize,
+    Metric::QueryCacheSize,
+    Metric::TaskClock,
+    Metric::TaskClockUser,
+    Metric::WorkProductIndexSize,
+];
+
 /// Compare 2 artifacts and print the result.
 pub async fn compare_artifacts(
     mut conn: Box<dyn Connection>,
-    base: String,
-    modified: String,
+    metric: Option<Metric>,
+    base: Option<String>,
+    modified: Option<String>,
 ) -> anyhow::Result<()> {
     let index = database::Index::load(&mut *conn).await;
 
-    let query = CompileBenchmarkQuery::default()
-        .metric(database::selector::Selector::One(Metric::InstructionsUser));
+    let metric = match metric {
+        Some(v) => v,
+        None => {
+            let metric_str = inquire::Select::new(
+                "Choose metric:",
+                ALL_METRICS.iter().map(|m| m.as_str()).collect::<Vec<_>>(),
+            )
+            .prompt()?;
+            Metric::from_str(metric_str).map_err(|e| anyhow::anyhow!(e))?
+        }
+    };
+
+    let mut aids = index.artifacts().map(str::to_string).collect::<Vec<_>>();
+    if aids.len() < 2 {
+        return Err(anyhow::anyhow!(
+            "There are not enough artifacts to compare, at least two are needed"
+        ));
+    }
+
+    let select_artifact_id = |name: &str, aids: &Vec<String>| {
+        anyhow::Ok(
+            inquire::Select::new(
+                &format!("Choose {} artifact to compare:", name),
+                aids.clone(),
+            )
+            .prompt()?,
+        )
+    };
+
+    let base = match base {
+        Some(v) => v,
+        None => select_artifact_id("base", &aids)?.to_string(),
+    };
+    if let Some(idx) = aids.iter().position(|id| id == &base) {
+        aids.remove(idx);
+    }
+    let modified = match modified {
```

Comment:
I have realized now that the common case locally will be that you have exactly two artifacts. If after filtering out `base` you have only one artifact left, we don't need to display the dialog and we can just select the only other remaining artifact.